fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/tree/TreeTypeHelper.java

```diff
@@ -0,0 +1,144 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.object.builder.v1.tree;
+
+import com.mojang.serialization.Codec;
+import net.fabricmc.fabric.mixin.object.builder.BlockStateProviderTypeInvoker;
+import net.fabricmc.fabric.mixin.object.builder.FoliagePlacerTypeInvoker;
+import net.fabricmc.fabric.mixin.object.builder.TreeDecoratorTypeInvoker;
+import net.fabricmc.fabric.mixin.object.builder.TrunkPlacerTypeInvoker;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.math.intprovider.IntProvider;
+import net.minecraft.util.math.intprovider.IntProviderType;
+import net.minecraft.world.gen.foliage.FoliagePlacer;
+import net.minecraft.world.gen.foliage.FoliagePlacerType;
+import net.minecraft.world.gen.stateprovider.BlockStateProvider;
+import net.minecraft.world.gen.stateprovider.BlockStateProviderType;
+import net.minecraft.world.gen.treedecorator.TreeDecorator;
+import net.minecraft.world.gen.treedecorator.TreeDecoratorType;
+import net.minecraft.world.gen.trunk.TrunkPlacer;
+import net.minecraft.world.gen.trunk.TrunkPlacerType;
+
+/**
+ * A registry for {@link TrunkPlacerType}s, {@link FoliagePlacerType}s, {@link TreeDecoratorType}s,
+ * {@link BlockStateProviderType}s, and {@link IntProviderType}s.
+ */
+public final class TreeTypeHelper {
+	/**
+	 * Creates a new instance of a {@link TrunkPlacerType}, registers and returns it.<br>
+	 * Uses {@link TrunkPlacerTypeInvoker} under the hood.
+	 *
+	 * <p>
+	 *    Example:<br>
+	 *
+	 *    {@code
+	 *          public static final TrunkPlacerType<MyTrunkPlacer> MY_TRUNK_PLACER =	FabricTreeRegistry.registerTrunkPlacerType(new Identifier("example", "my_trunk_placer"), MyTrunkPlacer.CODEC);
+	 *    }
+	 * </p>
+	 *
+	 * @param id Registry {@link Identifier}
+	 * @param codec The {@link Codec} associated with the {@link TrunkPlacer}
+	 * @param <T> The owner {@link TrunkPlacer}
+	 * @return Created and registered {@link TrunkPlacerType}
+	 */
+	public static <T extends TrunkPlacer> TrunkPlacerType<T> registerTrunkPlacerType(Identifier id, Codec<T> codec) {
+		return TrunkPlacerTypeInvoker.invokeRegister(id.toString(), codec);
+	}
+
+	/**
+	 * Creates a new instance of a {@link FoliagePlacerType}, registers and returns it.<br>
+	 * Uses {@link FoliagePlacerTypeInvoker} under the hood.
+	 *
+	 * <p>
+	 *     Example:<br>
+	 *
+	 *    {@code
+	 *         public static final FoliagePlacerType<MyFoliagePlacer> MY_FOLIAGE_PLACER = FabricTreeRegistry.registerFoliagePlacerType(new Identifier("tutorial", "my_foliage_placer"), MyFoliagePlacer.CODEC);
```

Comment:
This and other javadoc are using old refactored names.

FabricTreeRegistry -> TreeTypeHelper

Reply:
I'm dumb. Sorry